terraform/api_gateway.tf.json.template.py

```diff
@@ -219,12 +240,35 @@ def check_waf_rules(rules: JSONs) -> JSONs:
             for app in apps
         )
     ],
-    'locals': [
-        chalice.tf_config(app.name)['locals']
-        for app in apps
-    ],
+    'locals': list(chain(
+        [
+            chalice.tf_config(app.name)['locals']
+            for app in apps
+        ],
+        [
+            {
+                'user_ip': '${data.external.user_ip.result.ip}'
+            }
+        ]
+    )),
     'resource': [
         {
+            'aws_wafv2_ip_set': {
+                'it_v4_ips': {
+                    'name': config.qualified_resource_name('it_v4_ips'),
+                    'scope': 'REGIONAL',
+                    'ip_address_version': 'IPV4',
+                    'addresses': list(chain(
+                        [
+                            f'${{data.aws_nat_gateway.gitlab_{zone}.public_ip}}/32'
+                            for zone in range(vpc.num_zones)
+                        ],
+                        [
+                            '${local.user_ip}/32',
```

Comment:
What's the advantage of going through a script and a local? Couldn't we just directly make the IP check request here and interpolate it directly?

Reply:
Consider requesting PL for this.